client/landing/stepper/hooks/use-site-copy.tsx

```diff
@@ -0,0 +1,79 @@
+import { WPCOM_FEATURES_COPY_SITE } from '@automattic/calypso-products';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useMemo, useCallback } from 'react';
+import { useDispatch as useReduxDispatch, useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import { clearSignupDestinationCookie } from 'calypso/signup/storageUtils';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import {
+	hasLoadedSitePurchasesFromServer,
+	isFetchingSitePurchases,
+} from 'calypso/state/purchases/selectors';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+import { fetchSiteFeatures } from 'calypso/state/sites/features/actions';
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+function useSafeSiteHasFeature( siteId: number, feature: string ) {
+	const dispatch = useReduxDispatch();
+	useEffect( () => {
+		dispatch( fetchSiteFeatures( siteId ) );
+	}, [ dispatch, siteId ] );
+
+	return useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, feature );
+	} );
+}
```

Comment:
It would be great if we could import and reuse this hook. My concern is to avoid maintaining it in two places:

https://github.com/Automattic/wp-calypso/blob/trunk/client/sites-dashboard/components/sites-ellipsis-menu.tsx#L137-L146

The goal of this hook is similar to what `useQuerySitePurchases` does. "Fetching site data".

Reply:
I'm waiting for the marketing pr to merge in and clean it then. I have already done this locally :)